votor/src/certificate_pool.rs

```diff
@@ -87,6 +87,105 @@ impl From<AlpenglowCommitmentError> for AddVoteError {
     }
 }
 
+#[derive(Debug, Default)]
+struct FallbackVoteCounters {
+    my_first_vote: Option<Vote>,
+    total_stake: Stake,
+    skip_total: Stake,
+    notarize_total: Stake,
+    notarize_entry_total: BTreeMap<Hash, Stake>,
+    top_notarized_stake: Stake,
+    safe_to_notar_sent: Vec<Hash>,
+    safe_to_skip_sent: bool,
+}
+
+impl FallbackVoteCounters {
+    pub fn add_vote(
+        &mut self,
+        vote: &Vote,
+        total_stake: Stake,
+        entry_stake: Stake,
+        is_my_own_vote: bool,
+        events: &mut Vec<VotorEvent>,
+    ) -> (usize, bool) {
```

Comment:
would rather just pass in the mut stats struct here, so in the future we can increment more stats without polluting the return type.